include/nbl/system/IAsyncQueueDispatcher.h

```diff
@@ -73,7 +77,12 @@ class IAsyncQueueDispatcher : public IThreadHandler<CRTP, InternalStateType>, pu
         return x & Mask;
     }
 
+
 public:
+
+    IAsyncQueueDispatcher() {}
+    virtual ~IAsyncQueueDispatcher() {}
```

Comment:
get rid of virtual, why do you need them actually?

Reply:
because stuff inherits from `IAsyncQueueDispatcher` and the correct destructors dont get called in the right order?

Reply:
ah yeah I see the comment about the class not being meant for runtime polymorphism